Detour/Source/DetourNavMeshQuery.cpp

```diff
@@ -2070,13 +2101,17 @@ dtStatus dtNavMeshQuery::moveAlongSurface(dtPolyRef startRef, const float* start
 	dtAssert(m_nav);
 	dtAssert(m_tinyNodePool);
 
-	*visitedCount = 0;
-	
 	// Validate input
-	if (!startRef)
-		return DT_FAILURE | DT_INVALID_PARAM;
-	if (!m_nav->isValidPolyRef(startRef))
+	if (!m_nav->isValidPolyRef(startRef) ||
+		!startPos || !dtVisfinite(startPos) ||
+		!endPos || !dtVisfinite(endPos) ||
+		!filter || !resultPos || !visited || !visitedCount ||
+		maxVisitedSize <= 0)
+	{
 		return DT_FAILURE | DT_INVALID_PARAM;
+	}
+
+	*visitedCount = 0;
```

Comment:
I used to have rule that output should always be valid (i.e. setting return array counts to zero), even when returning failure. I think that still held after `dtStatus` was introduced. Might not have been consistently implemented though. Some code might rely on that.

Reply:
Will make sure this is still the case then.